nntrainer/layers/layer_node.cpp

```diff
@@ -351,6 +351,8 @@ void LayerNode::finalize() {
   if (finalized)
     throw std::runtime_error("Finalizing a layer which is already finalized");
 
+  init_context = InitLayerContext(init_context.getInputDimensions(),
+                                  init_context.getNumOutputs(), getName());
```

Comment:
I wonder layer_node has name and init_context has same name. So.. it might be possible to remove one of them?

Reply:
`init_context` provides access to the name of the layer node to its underneath Layer::finalize method.

Reply:
if it is, then great. Thanks.